Documentation: Add comprehensive project documentation and contribution guidelines (#34)
FiligranHQ/xtm-composer: README.md

# XTM Composer

The following repository is used to store the platform XTM composer.
The composer allows OpenCTI / OpenBAS users to manage their connectors/collectors/injectors directly from the platform.
For performance and low level access, the agent is written in Rust. Please start your journey
with https://doc.rust-lang.org/book.

## Documentation

- [Installation Guide](docs/installation.md) - System requirements and installation methods
- [Quick Start](docs/quick-start.md) - Get up and running quickly
- [Configuration Reference](docs/configuration.md) - Complete configuration documentation
- [Development Guide](docs/development.md) - Setup for development and contribution

## Module Status

- **OpenCTI**: ✅ Fully implemented and production-ready

> - **OpenBAS**: 🚧 Coming Soon

Reviewer: ```suggestion
- **OpenAEV**: 🚧 Coming Soon
```

1. Change version in cargo.toml (ex:1.0.0)
2. Push that version on master
3. Create a git tag with same numbers (ex:1.0.0)

> 4. Push the git tag with a format X.X.X on master branch: the docker image is build with this tag too.

Reviewer 2: ```suggestion
4. Push a git tag in the format X.X.X to the master branch. The Docker image will be built with the same tag.
```
